news/2024/2024-09-17-performance-points-star-rating-updates.md

```diff
@@ -0,0 +1,175 @@
+
+Put another way, missing in long beatmaps with a single "difficulty spike" will be punished harder than in long maps that are consistently difficult. The algorithm essentially assumes that misses happen on the most difficult parts.
+
+As touched on earlier, big jumps in total pp are expected in this round. The removal of combo scaling will result in large increases for users who have many scores with lower combos that are considered difficult. It may also result in decreases for users who have many scores with higher combos but saturated a lot of misses towards the end of a beatmap.
+
+### Rhythm complexity improvements
+
+Rhythm complexity can be expressed as "changes in pattern lengths and timings" — quintuples changing into triples, even bursts into odd bursts, 1/4 streams into 1/3 streams, etc. More erratic changes imply a more complex rhythm, making it harder for players to adapt and recognise the rhythm they should tap to.
+
+A [change](https://github.com/ppy/osu/pull/28871) proposed by [StanR](https://osu.ppy.sh/users/7217455) fixes some flaws in osu!'s rhythm complexity calculations:
+
+- Highly complex rhythm has received a general buff, affecting maps such as [II-L - Asteroid Field of DECAPLETS](https://osu.ppy.sh/beatmapsets/1666685#osu/3403124).
+- A bonus was previously applied based on pattern length, which caused long bursts to be assessed as rhythmically complex even if they were not. This has been removed.
+- "Double-tappable" patterns have been nerfed, i.e. overlapping notes where the time between objects is low enough to comfortably double-tap and still receive 300s. This double-tap nerf already existed in the speed portion of pp, and this change now brings it over to rhythm complexity calculations too.
+- Repeated note groups have been heavily nerfed. This practically means that patterns such as repeated triple spam (like in [nameless - SLoWMoTIoN](https://osu.ppy.sh/beatmapsets/295848#osu/988793)) are nerfed significantly:
+
+![](/wiki/shared/news/2024-09-17-performance-points-star-rating-updates/island-repetition-nerf.png)
+
+### Speed distance hotfix
+
+A [change](https://github.com/ppy/osu/pull/29980) proposed by [tsunyoku](https://osu.ppy.sh/users/11315329) brings a minor change to a bonus applied for speed distance bonus.
+
+The distance bonus exists to reward flow aim by multiplying it with the speed difficulty. In this hotfix, rather than the bonuses being multiplicative, they have been changed to be additive to decrease the effect the distance bonus has on the speed value.
+
+In addition, the distance bonus scaling has been changed to decrease the reward for lower-spacing flow aim. Finally, a new multiplier was added to the final result of the distance bonus to decrease it overall.
+
+![](/wiki/shared/news/2024-09-17-performance-points-star-rating-updates/distance-bonus.png)
```

Comment:
wouldn't it be way better to illustrate the [multiplicative-to-additive change](https://github.com/ppy/osu/pull/29980/files#diff-63100aca8fa85bf402677740e45b3fc1ab11c505357c859740bf591b35cacf98L70-R71) in graph form than the [change to the distance multiplier](https://github.com/ppy/osu/pull/29980/files#diff-63100aca8fa85bf402677740e45b3fc1ab11c505357c859740bf591b35cacf98L66-R68)? isn't it way more of an impactful change?

Reply:
in that case i'd leave the graph out completely. showing that the "low ar distance bonus" is halved (appearance-wise, because the y axis starts at 0.25 or smth) gives no better intuition than the text description since the changes aren't in terms of actual pp values

Reply:
hm, I was thinking with the circle distance being the x-axis that it would give a good understanding of how the distance bonus scales even if that doesn't necessarily translate to pp directly - is it not intuitive to look at the graph and think "flow aim that is smaller than *x* circles distance is going to give a smaller bonus than before"? it just seems like a visualization of the explanation to me. the issue with the additive change was that you don't get this same intuition imo

Reply:
the graph shows two minimally adjusted curves that *do* need explanation, but what it's depicting is just a small part in the equation that i think is explained well enough in the revised text i suggested. it depicts a change in the code really, not what its effects are (i certainly didn't have a clue what the graph was about before reading the code)

will people think it's a small adjustment because of the small visual difference between the curves? will people think the pp for low spacing is halved since the y axis misleadingly starts at 0.25? will people understand how much and in what cases this "multiplier" affects things anyway? will people understand that the graph only depicts the second, smaller change in the first place? the circle distance thing on the x axis might be understandable to a larger proportion of readers, but the rest isn't

"lower-spacing fast flow aim is nerfed" is just about what needs to be conveyed for the `distanceBonus` change, and in general the combined effect is a sizeable nerf to flow aim for speed players - [pishi's video](https://www.youtube.com/watch?v=Kqu4TUOO5IY&t=155s) explains this well, doesn't it?

a good way to show the effects of the pr in question might be to mention which maps get the largest change in pp and by how much (we have [spreadsheets for this](https://github.com/ppy/osu/pull/29980#issuecomment-2396025182) don't we). something like "most of the affected >1k pp plays lose about 40 to 70 pp from this change". maybe graphing this spreadsheet would be a better idea

Reply:
in the past we've given some examples of SR/PP changes on some maps that were effected by changes, I'll do the same here and remove the distance bonus graph